### DISCUSSION TITLE
How to have next JS read from an env file in another directory?

### DISCUSSION THREAD

**Discussion by ian-whitestone** [vercel/next.js#39955] [Help] [answered]
My github repository file system looks like this:

```
.
├── web/
│   ├── frontend/
│   │   ├── package.json
│   │   ├── next.config.js
│   │   └── src/
│   │       ├── pages/
│   │       ├── components/
│   │       └── ...
│   └── backend/
│       └── app.py
├── other_stuff/
│   └── more_things/
│       └── ...
├── .env.local
├── README.md
└── ...
```

To run my next dev server, I navigate to `web/frontend/` then run `next dev`. This doesn't detect/load the `.env.local` file at the root of my repository. I instead have to copy the `.env.local` file to the `web/frontend/` folder. This is not ideal for me, since this `.env.local` file is also used by other processes (python server in web/backend). 

Is there anyway to have next JS look for `.env` files that are in another directory?

Any help or advice would be greatly appreciated!

**Comment by icyJoseph** [comment]
Hi,

I followed this article: https://medium.com/courtly-intrepid/environmental-variables-in-next-js-with-dotenv-599c5bbfdf74, and it worked fine, **_but with one major caveat. All env vars added through this method ended up on the browser, even if NOT prefixed with `NEXT_PUBLIC`. Env vars loaded by Next.js as usual, are removed from the client, unless prefixed with `NEXT_PUBLIC`._**

```js
const webpack = require("webpack");

const { parsed: myEnv } = require("dotenv").config({
  path: "./envs/local",
});

/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  webpack(config) {
    // all vars end up in the client -
    config.plugins.push(new webpack.EnvironmentPlugin(myEnv));
    return config;
  },
};

module.exports = nextConfig;
```

**Reply by ian-whitestone** [reply]
Thanks @icyJoseph , appreciate the response. I ended up going with the Symlink approach so I could keep using the `NEXT_PUBLIC` env feature.

**Reply by icyJoseph** [reply]
Yeah, I'm also saving that option for later! 😂

**Comment by miguderp** [comment, marked as answer]
You can also simply create a Symlink (symbolic link, [more info](https://kb.iu.edu/d/abbe));

`ln -s <where file is actually located> <where you want to create a shortcut>`

Which in your case means (if I understood correctly):
`ln -s ./.env.local ./web/frontend/.env.local`

**Reply by ian-whitestone** [reply]
Thanks @miguderp !

I ended up having to do `ln -s $PWD/.env.local ./web/frontend/.env.local`, for some reason it wasn't happy with just `./.env.local`, but this did the trick!
